petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -38,10 +38,30 @@ def test_primitive(self):
             tensor = iterator.get_next()
             with tf.Session() as sess:
                 ts = sess.run(tensor)
+                try:
+                    sess.run(iterator.get_next())
+                except tf.errors.OutOfRangeError:
+                    self.fail("The tf.data.Dataset cannot read all the rows from the spark dataframe.")
```

Comment:
To read all rows, you need a loop:
```
next_tensor = iterator.get_next()
results = []
for i in range(batch_count):
    results.append(sess.run(next_tensor))
```
Then check the results with original spark dataframe.

Reply:
The `batch_count` would be 1 if we test with small df since the `parquet.block.size` is large (I tried to set it to 250 (WARN InternalParquetRecordWriter: Too much memory used: Store) and 500 (No warning) but neither make the `batch_count == 2`). If we use `numPartition > 1` to let `batch_size == numPartitions`, the order of the loaded dataset would be non-deterministic.